test/e2e/secret.py

```diff
@@ -0,0 +1,42 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Fixtures common to all RDS controller tests"""
+
+import dataclasses
+
+from acktest.k8s import resource as k8s
+
+import logging
+
+from dataclasses import dataclass, field
+
+from acktest import bootstrapping
+
+
+@dataclasses.dataclass
+class Secret(bootstrapping.Bootstrappable):
+    ns: str
+    name: str
+    key: str
+    val: str
+
+    def bootstrap(self):
+        """Ensures a Kubernetes secret with the specified ns/name/key exists
+        """
+        k8s.create_opaque_secret(self.ns, self.name, self.key, self.val)
+
+    def cleanup(self):
+        """Ensures the Kubernetes secret does not exist
+        """
+        k8s.delete_secret(self.ns, self.name)
```

Comment:
I think this is better suited to a Pytest fixture. There is no need for this to be bootstrapped separately from the tests, as it only creates resources within the cluster.

Reply:
Definitely to be looked at in a future PR. I tried to go the pytest fixture route and ended up tearing all my hair out. Well, I didn't have any hair to begin with, so started gouging my eyeballs out and that was too painful.